MySQLdb/connections.py

```diff
@@ -152,30 +160,31 @@ class object, used to create cursors (keyword only)
                 conv2[k] = v[:]
             else:
                 conv2[k] = v
-        kwargs2['conv'] = conv2
-
-        cursorclass = kwargs2.pop('cursorclass', self.default_cursor)
-        charset = kwargs2.get('charset', '')
-        use_unicode = kwargs2.pop('use_unicode', True)
-        sql_mode = kwargs2.pop('sql_mode', '')
-        self._binary_prefix = kwargs2.pop('binary_prefix', False)
-
-        client_flag = kwargs.get('client_flag', 0)
-        client_version = tuple([ numeric_part(n) for n in _mysql.get_client_info().split('.')[:2] ])
+        kwargs2["conv"] = conv2
+
+        cursorclass = kwargs2.pop("cursorclass", self.default_cursor)
+        charset = kwargs2.get("charset", "")
+        use_unicode = kwargs2.pop("use_unicode", True)
+        sql_mode = kwargs2.pop("sql_mode", "")
+        self._binary_prefix = kwargs2.pop("binary_prefix", False)
+
+        client_flag = kwargs.get("client_flag", 0)
+        client_version = tuple(
+            [numeric_part(n) for n in _mysql.get_client_info().split(".")[:2]]
+        )
         if client_version >= (4, 1):
             client_flag |= CLIENT.MULTI_STATEMENTS
         if client_version >= (5, 0):
             client_flag |= CLIENT.MULTI_RESULTS
 
-        kwargs2['client_flag'] = client_flag
+        kwargs2["client_flag"] = client_flag
 
         # PEP-249 requires autocommit to be initially off
-        autocommit = kwargs2.pop('autocommit', False)
+        autocommit = kwargs2.pop("autocommit", False)
 
-        super(Connection, self).__init__(*args, **kwargs2)
+        super().__init__(*args, **kwargs2)
         self.cursorclass = cursorclass
-        self.encoders = dict([ (k, v) for k, v in conv.items()
-                               if type(k) is not int ])
+        self.encoders = {k: v for k, v in conv.items() if not isinstance(k, int)}
```

Comment:
This is intentional, don't change it.

```suggestion
        self.encoders = {k: v for k, v in conv.items() if type(k) is not int}
```